salt/modules/debbuild.py

```diff
@@ -416,20 +442,66 @@ def make_repo(repodir, keyid=None, env=None):
     with salt.utils.fopen(repoconfdist, 'w') as fow:
         fow.write('{0}'.format(repocfg_dists))
 
+    local_fingerprint = None
+    local_keyid = None
+
     if keyid is not None:
         with salt.utils.fopen(repoconfdist, 'a') as fow:
             fow.write('SignWith: {0}\n'.format(keyid))
 
+        # gpg keys should have been loaded as part of setup
+        # retrieve specified key, obtain fingerprint and preset passphrase
+        local_keys = __salt__['gpg.list_keys'](user=runas, gnupghome=gnupghome)
+        for gpg_key in local_keys:
+            if keyid == gpg_key['keyid'][8:]:
+                local_fingerprint = gpg_key['fingerprint']
+                local_keyid = gpg_key['keyid']
+                break
+
+        if local_keyid is None:
```

Comment:
Here, local_keyid will be undef if nothing in the loop matches and therefore will raise a `NameError`.